blocks/page_list/templates/itinerary/Itinerary-Store.js

```diff
@@ -0,0 +1,64 @@
+import { dispatch, register } from './Itinerary-Dispatcher';
+import { EventEmitter } from 'events';
+import ItineraryConstants from './Itinerary-Constants';
+import walks from './Itinerary-Static-Data';
+
+const CHANGE_EVENT = 'change';
+
+let _walks = {walks:walks.slice(0,-1)}; //grabbing static walks for now
+
+const _removeWalk = ( id ) => {
+  _walks.splice( _walks.findIndex(walks => walks.id === id),1);
+};
+
+
+const _addWalk = ( id ) => {
+  let walkExists = _walks.walks.findIndex(walk => walk.id === id);
+
+  if(walkExists === -1) {
+    let walkToAdd = walks.filter(walk => walk.id === id)[0];
```

Comment:
Add `Array.prototype.find` to the `<root>/js/shims.js`, and use here instead of filter.